tests/test_protocols/test_generator/test_extract_specification.py

```diff
@@ -0,0 +1,533 @@
+
+        specification_type_3 = (
+            "pt:union[ct:DataModel, pt:set[pt:int], pt:list[pt:bool], pt:bytes, pt:dict[pt:bool,pt:float], pt:int, "
+            "pt:set[pt:bool], pt:dict[pt:int, pt:str], pt:list[pt:str], pt:bool, pt:float, pt:str, pt:dict[pt:str, pt:str]]"
+        )
+        expected_3 = (
+            "Union[DataModel, FrozenSet[int], Tuple[bool, ...], bytes, Dict[bool, float], int, "
+            "FrozenSet[bool], Dict[int, str], Tuple[str, ...], bool, float, str, Dict[str, str]]"
+        )
+        assert _mt_specification_type_to_python_type(specification_type_3) == expected_3
+
+    def test_optional_specification_type_to_python_type(self):
+        """Test the '_optional_specification_type_to_python_type' method."""
+        specification_type_1 = (
+            "pt:optional[pt:union[ct:DataModel, pt:bytes, pt:int, pt:bool, pt:float, pt:str, pt:set[pt:int], "
+            "pt:list[pt:bool], pt:dict[pt:str, pt:str]]]"
+        )
+        expected_1 = "Optional[Union[DataModel, bytes, int, bool, float, str, FrozenSet[int], Tuple[bool, ...], Dict[str, str]]]"
+        assert (
+            _optional_specification_type_to_python_type(specification_type_1)
+            == expected_1
+        )
+
+        specification_type_2 = (
+            "pt:optional[pt:union[ct:DataModel, pt:bytes, pt:int, pt:bool, pt:float, pt:str, pt:set[pt:int], "
+            "pt:list[pt:bool], pt:dict[pt:str,pt:str]]]"
+        )
+        expected_2 = "Optional[Union[DataModel, bytes, int, bool, float, str, FrozenSet[int], Tuple[bool, ...], Dict[str, str]]]"
+        assert (
+            _optional_specification_type_to_python_type(specification_type_2)
+            == expected_2
+        )
+
+        specification_type_3 = "pt:optional[ct:DataModel]"
+        expected_3 = "Optional[DataModel]"
+        assert (
+            _optional_specification_type_to_python_type(specification_type_3)
+            == expected_3
+        )
+
+    def test_specification_type_to_python_type(self):
+        """Test the '_specification_type_to_python_type' method."""
+        specification_type_1 = "ct:DataModel"
+        expected_1 = "DataModel"
+        assert _specification_type_to_python_type(specification_type_1) == expected_1
+
+        specification_type_2 = "pt:bytes"
+        expected_2 = "bytes"
+        assert _specification_type_to_python_type(specification_type_2) == expected_2
+
+        specification_type_3 = "pt:set[pt:int]"
+        expected_3 = "FrozenSet[int]"
+        assert _specification_type_to_python_type(specification_type_3) == expected_3
+
+        specification_type_4 = "pt:list[pt:float]"
+        expected_4 = "Tuple[float, ...]"
+        assert _specification_type_to_python_type(specification_type_4) == expected_4
+
+        specification_type_5 = "pt:dict[pt:bool, pt:str]"
+        expected_5 = "Dict[bool, str]"
+        assert _specification_type_to_python_type(specification_type_5) == expected_5
```

Comment:
This would be a great usecase for fixtures

Reply:
I don't know about fixtures. I will follow up.